tmk_core/protocol/ps2_interrupt.c

```diff
@@ -232,40 +261,62 @@ static uint8_t     pbuf[PBUF_SIZE];
 static uint8_t     pbuf_head = 0;
 static uint8_t     pbuf_tail = 0;
 static inline void pbuf_enqueue(uint8_t data) {
+#if defined(__AVR__)
     uint8_t sreg = SREG;
     cli();
+#endif
+
     uint8_t next = (pbuf_head + 1) % PBUF_SIZE;
     if (next != pbuf_tail) {
         pbuf[pbuf_head] = data;
         pbuf_head       = next;
     } else {
         print("pbuf: full\n");
     }
+
+#if defined(__AVR__)
     SREG = sreg;
+#endif
 }
 static inline uint8_t pbuf_dequeue(void) {
     uint8_t val = 0;
 
+#if defined(__AVR__)
     uint8_t sreg = SREG;
     cli();
+#endif
```

Comment:
Here, the chibios analog would be `chSysLock` or `osalSysLock`, since this is called from the main thread.